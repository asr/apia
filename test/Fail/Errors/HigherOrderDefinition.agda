------------------------------------------------------------------------------
-- We only translate first-order definitions
------------------------------------------------------------------------------

{-# OPTIONS --exact-split              #-}
{-# OPTIONS --no-sized-types           #-}
{-# OPTIONS --no-universe-polymorphism #-}
{-# OPTIONS --without-K                #-}

------------------------------------------------------------------------------

postulate
  D    : Set
  _≡_  : D → D → Set
  succ : D → D

-- A higher-order definition
twice : (D → D) → D → D
twice f x = f (f x)
{-# ATP definition twice #-}

postulate twice-succ : ∀ n → twice succ n ≡ succ (succ n)
{-# ATP prove twice-succ #-}
