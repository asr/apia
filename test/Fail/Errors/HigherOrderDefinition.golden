ret > ExitFailure 1
err > apia: the translation of ‘HigherOrderDefinition.twice’ failed because it is not a FOL-definition
err >
